semgrep/tests/unit/test_semgrep_test.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+from itertools import chain
+from itertools import combinations
+from itertools import product
+from typing import Iterator
+from typing import Set
+from typing import Tuple
+
+import pytest
+
+from semgrep.test import line_has_ok
+from semgrep.test import line_has_rule
+from semgrep.test import normalize_rule_ids
+
+# cf. https://docs.python.org/3/library/itertools.html#itertools-recipes
+def powerset(iterable):
+    "powerset([1,2,3]) --> () (1,) (2,) (3,) (1,2) (1,3) (2,3) (1,2,3)"
+    s = list(iterable)
+    return chain.from_iterable(combinations(s, r) for r in range(len(s) + 1))
+
+
+COMMENT_SYNTAXES = (("#", "\n"), ("//", "\n"), ("<!--", "-->"), ("(*", "*)"))
```

Comment:
It seems overly fragile to be redefining these here. I.e. now we'll have to update two places when updating this information. Can we define these in one place, like `test.py`, and use them from there?

IMO we should define the language comment syntax's and autogenerate all the annotations from there. E.g. `['#', '//', '(*', ...]` -> `['# ruleid:', '// ruleid:', ..., '# ok:', '// ok:', ...]`. Then we'll also be less likely to forget annotation definitions for certain comment syntax's.

Reply:
Yeah, that's a good idea 👍

Reply:
Done!